### PR TITLE
feat(aio): docs page revised and version selector added to sidenav

### PR DESCRIPTION
**Pull request by wardbell**
@juleskremer provided new text and format and asked for a version selector at the bottom of the left sidenav.

Also deletes dead news page.

**Remember to run `yarn docs` locally as this PR changes docs and `navigation.json`.**

Testing of these changes led to refactoring of tests for dependent services.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by wardbell** [comment]
I fully expect a merge conflict on `location.service.spec.ts` after earlier merge-ready PRs land. Please let's land those so I can clean this up.

File: aio/content/navigation.json [hunk 1]
```diff
@@ -482,5 +485,10 @@
         }
       ]
     }
+  ],
+
+  "__docVersions": [
```
Comment (petebacondarwin) [inline aio/content/navigation.json]: Why the double underscore. This is still a navigation view. It just happens that we choose to render it differently to the side nav. But we already do a similar custom rendering for the Top bar.
I don't see why we need to indicate it is private.

File: aio/src/app/navigation/navigation.model.ts [hunk 2]
```diff
@@ -28,6 +28,11 @@ export interface CurrentNode {
   nodes: NavigationNode[];
 }
 
+export interface DocVersion {
+  name: string;
+  href?: string;
+}
```
Comment (petebacondarwin) [inline aio/src/app/navigation/navigation.model.ts]: Why create a new interface for this?
`NavigationNode` already provides exactly what you need (`title` and `url`).

File: aio/src/app/app.component.html [hunk 3]
```diff
@@ -17,8 +17,8 @@
     <aio-nav-menu [nodes]="sideNavNodes" [currentNode]="currentNode" ></aio-nav-menu>
 
     <div class="doc-version" title="Angular docs version {{currentDocVersion?.name}}">
-      <select #ver [value]="currentDocVersion" (change)="onDocVersionChange(ver.selectedIndex)">
-        <option *ngFor="let version of docVersions" [value]="version">{{version.name}}</option>
+      <select (change)="onDocVersionChange($event.target.selectedIndex)">
```
Comment (petebacondarwin) [inline aio/src/app/app.component.html]: +1

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
The changes to `LocationService` and its specs look great!
Also I like the DRYing up of the `NavigationService` specs.
The front end rendering of the docs versions looks good. I'm glad you managed to remove the need for the template reference variable.

I don't feel that we need to add in a new concept for the `DocsVersions` though. We should just make it another navigation view, reusing all of the machinery that is in place for that already.

**Comment by wardbell** [comment]
Expected merge conflicts resolved. @petebacondarwin suggestions incorporated. Ready for review.

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
It's a shame the select is so ugly - perhaps we could make it more like the selectors on the API filter page in a future PR?

**Comment by wardbell** [comment]
I deliberately didn't style it for this PR as I was asked to make it minimal and leave styling to @sjtrimble. @juleskremer to look at it and decide what's best.

**Pull request merged**